tools/analyzer_plugin/test/integration/diagnostics/duplicate_prop_cascade_test.dart

```diff
@@ -0,0 +1,69 @@
+// Disable null-safety in the plugin entrypoint until all dependencies are null-safe,
+// otherwise tests won't be able to run. See: https://github.com/dart-lang/test#compiler-flags
+// @dart=2.9
+import 'dart:async';
+
+import 'package:over_react_analyzer_plugin/src/diagnostic/duplicate_prop_cascade.dart';
+import 'package:test/test.dart';
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../test_bases/diagnostic_test_base.dart';
+
+void main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(DuplicatePropCascadeDiagnosticTest);
+  });
+}
+
+@reflectiveTest
+class DuplicatePropCascadeDiagnosticTest extends DiagnosticTestBase {
+  @override
+  get errorUnderTest => DuplicatePropCascadeDiagnostic.code;
+
+  @override
+  get fixKindUnderTest => DuplicatePropCascadeDiagnostic.fixKind;
+
+  Future<void> test_noError() async {
+    final source = newSource('test.dart', /*language=dart*/ r'''
+part 'test.over_react.g.dart';
+
+mixin CustomProps on UiProps {
+  int size;
+  bool hidden;
+}
+
+final Custom = uiFunction<CustomProps>(
+  (props) {
+    return null;
+  },
+  _$CustomConfig, // ignore: undefined_identifier
+);
+
+final customUsage = (Custom()
+  ..size = 2
+  ..dom.size = 2
+  ..dom.hidden = false
+  ..aria.hidden = false
+  ..hidden = false
```

Comment:
This test covers a lot of cases, but it might not be clear to future maintainers what this is testing and why. I think it'd be good to have a comment or a better test name that documents all of the cases we want to hit (e.g., props from same class but with a prefix, props that have the same name but aren't from the same class, etc.).

Having them as separate tests might be nice, but I'm also totally fine having them be all in one so long as the intent of the test is clear.

Same comment for the test below.

Also, I noticed that there isn't a test for the same prop being provided twice, without any prefix, which is the much more common case we want to make sure is handled. E.g.,
```dart
final foo = (Dom.div()
  ..id = '1'
  ..id = 'foo'
)();
```
Could we add a test case for that, and also one for more than two usages?